bson/src/main/org/bson/codecs/pojo/ConventionSetPrivateFieldImpl.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.pojo;
+
+import org.bson.codecs.configuration.CodecConfigurationException;
+
+import java.lang.reflect.Field;
+
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isPrivate;
+
+final class ConventionSetPrivateFieldImpl implements Convention {
+
+    @Override
+    public void apply(final ClassModelBuilder<?> classModelBuilder) {
+        for (PropertyModelBuilder<?> propertyModelBuilder : classModelBuilder.getPropertyModelBuilders()) {
+            if (propertyModelBuilder.getPropertyAccessor() instanceof PropertyAccessorImpl) {
```

Comment:
I have some concern about a silent failure to use this convention when combined with other conventions that are using some other subclass of PropertyAccessor.   Since it's unlikely to happen often in practice and would be quite confusing when it does happen, we should throw an exception here instead of just falling through.  Something like:

    throw new CodecConfigurationException("SET_PRIVATE_FIELDS_CONVENTION not compatible with conventions that use their own implementations of org.bson.codecs.pojo.PropertyAccessor: " + propertyModelBuilder.getPropertyAccessor().getClass().getName())